src/inewsFtp.ts

```diff
@@ -0,0 +1,653 @@
+					callbackSafe(null, result!)
+				}
+				next()
+			})
+		}
+
+		const timedOperation = (callback: NodeCallback<T>) => {
+			this._timedOperation<Operation<T>, T>(operation, operationTimeout, (error, result) => {
+				if (!operationComplete) {
+					// Only continue if not canceled
+					if (error) {
+						callback(error)
+					} else {
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						callback(null, result!)
+					}
+				}
+			})
+		}
+
+		const callbackSafe: NodeCallback<T> = (error, result) => {
+			if (!operationComplete) {
+				operationComplete = true
+				if (typeof finalCallback === 'function') {
+					if (error) {
+						finalCallback(error)
+					} else {
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						finalCallback(null, result!)
+					}
+				}
+			}
+		}
+
+		const jobController = this._queue.enqueue(jobOperation, options)
+
+		return {
+			cancel: () => {
+				operationComplete = true
+				jobController.cancel()
+			},
+			complete: () => {
+				operationComplete = true
+				jobController.complete()
+			},
+			restart: () => {
+				jobController.restart()
+			},
+		}
+	}
+
+	private _attemptOperation<T>(
+		operation: Operation<T>,
+		maxAttempts: number,
+		errorCallback: ErrorCallback,
+		finalCallback?: NodeCallback<T>
+	) {
+		let currentAttempt = 0
+
+		const attemptOperation = () => {
+			currentAttempt++
+			const operationAttempt = currentAttempt
+			operation((error, result) => {
+				if (error && (typeof maxAttempts !== 'number' || maxAttempts < 0 || currentAttempt < maxAttempts)) {
+					errorCallback(error, (continueError) => {
+						if (continueError) callbackSafe(operationAttempt, continueError, result)
+						else attemptOperation()
+					})
+				} else callbackSafe(operationAttempt, error, result)
+			})
+		}
+
+		const callbackSafe = (
+			operationAttempt: number,
+			error: NodeJS.ErrnoException | null | undefined,
+			result: T | null | undefined
+		) => {
+			if (operationAttempt === currentAttempt) {
+				if (typeof finalCallback === 'function') {
+					if (error) {
+						finalCallback(error)
+					} else {
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						finalCallback(null, result!)
+					}
+				}
+			}
+		}
+
+		attemptOperation()
+	}
+
+	private _timedOperation<T extends Operation<U>, U>(operation: T, timeout: number, callback: NodeCallback<U>) {
+		const operationTimeout = setTimeout(() => {
+			callback(new Error('operation_timeout'), null)
+		}, timeout)
+
+		operation((...args: any[]) => {
+			clearTimeout(operationTimeout)
+			callback(args[0], args[1])
+		})
+	}
+
+	private _objectMerge(...args: Array<Record<string, unknown>>) {
+		const merged = {}
+		this._objectForEach(args, (argument: Record<string, unknown>) => {
+			for (const attrname in argument) {
+				if ({}.hasOwnProperty.call(argument, attrname)) (merged as any)[attrname] = argument[attrname]
+			}
+		})
+		return merged
+	}
```

Comment:
By supplying the return type of the function along with a type to `merged`, casting to `any` can be avoided:
```suggestion
	private _objectMerge(...args: Array<Record<string, unknown>>): Record<string, unknown> {
		const merged: Record<string, unknown> = {}
		this._objectForEach(args, (argument: Record<string, unknown>) => {
			for (const attrname in argument) {
				if ({}.hasOwnProperty.call(argument, attrname)) {
					merged[attrname] = argument[attrname]
				}
			}
		})
		return merged
	}
```
The same approach can be used for some of the other places where indexable constructs have been cast to `any`.